lib/cchain/provider/abci.go

```diff
@@ -373,7 +373,17 @@ func searchOffsetInHistory(ctx context.Context, client rpcclient.Client, cl atyp
 		endHeightIndex = queryHeight
 		lookback *= 2
 		if queryHeight < lookback {
-			startHeightIndex = 1
+			earliestStoreHeight, err := getEarliestStoreHeight(ctx, cl, chainVer, 1)
```

Comment:
what about rather setting `queryHeight=1` and looping back to `for queryHeight > 0 {`?

no additional logic required in that case, we rerun existing logic.

Reply:
Also nit: `for queryHeight > 0` is always true, not sure that condition is actually every triggered? can just do `for {` instead?

Reply:
Yup, both of those seem reasonable to me, thanks!